orm/index.go

```diff
@@ -183,33 +216,61 @@ func (i Index) move(db weave.KVStore, prev Object, save Object) error {
 		return ErrModifiedPK()
 	}
 
-	// if the keys don't change, then
-	oldKey, err := i.index(prev)
+	oldKeys, err := i.index(prev)
 	if err != nil {
 		return err
 	}
-	newKey, err := i.index(save)
+	newKeys, err := i.index(save)
 	if err != nil {
 		return err
 	}
-	if bytes.Equal(oldKey, newKey) {
-		return nil
+
+	oldKeysSet := make(map[string]struct{})
+	for _, v := range oldKeys {
+		oldKeysSet[string(v)] = struct{}{}
 	}
 
-	// check unique constraint before removing
-	if i.unique && len(oldKey) != 0 {
-		k := i.IndexKey(newKey)
-		val := db.Get(k)
-		if val != nil {
-			return ErrUniqueConstraint(i.name)
+	newKeysSet := make(map[string]struct{})
+	for _, v := range newKeys {
+		newKeysSet[string(v)] = struct{}{}
+	}
+
+	// check unique constraints first
+	for _, newKey := range newKeys {
+		if _, exists := oldKeysSet[string(newKey)]; exists { // skip existing keys
+			continue
+		}
+		// check unique constraint before removing
+		if i.unique && len(oldKeysSet) != 0 {
```

Comment:
why is this `len(oldKeysSet) != 0` check there?

maybe this is important? maybe there was some odd logic in my old code?
good to add comment if we really want it

Reply:
I stumbled upon this, too. In the original code an old key must exists to execute the unique check here. I tried to mimic this behaviour with the `len` check. anyway, I am more than happy to remove the extra condition.